firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -589,6 +589,7 @@ class BrowserCoordinator: BaseCoordinator,
             navigationController.modalPresentationStyle = .pageSheet
             navigationController.sheetPresentationController?.detents = [.medium(), .large()]
             navigationController.sheetPresentationController?.prefersGrabberVisible = true
+            store.dispatch(ToolbarAction(windowUUID: windowUUID, actionType: ToolbarActionType.cancelEdit))
```

Comment:
I don't think this is the best solution from a usability perspective: imagine the user has typed their search term already and then wants to switch search engines. Now with us canceling edit mode they lost everything they typed and have to start over entering the search term. 
I think it would be better to just dismiss the keyboard but stay in edit mode.

Reply:
Yes, you're right. I believe that that was the behavior before, but by rotating the device while in edit mode, triggers the keyboard.

Reply:
Ahhh, so THAT is what's triggering the keyboard to reappear? 🤔  Good catch @PARAIPAN9 , thanks for looking into this for me! 🙏 

@PARAIPAN9 If we can't use cancelEdit, maybe we need to do something similar as AddressBarState's `isScrollingDuringEdit`, which when set to true hides the keyboard on line 268 of LocationView:

```
        let shouldShowKeyboard = state.isEditing && !state.isScrollingDuringEdit
```

So maybe we can add another state property so `shouldShowKeyboard` returns false (or at least doesn't get reset to true) when we rotate AND the search engine bottom sheet is visible. Possibly we need to set something on the redux state every time the search engine bottom sheet is shown, and then clear it when it is dismissed. 🧐

Reply:
Yes @ih-codes, this seems like a good approach.